### PR TITLE
Created an initial Localstack commodity

### PR DESCRIPTION
**Pull request by brooks630**
* **What kind of change does this PR introduce (Bug fix, feature, docs update, ...)?**
  Feature

* **What is the current behavior?**
  Applications are currently provisioning their own Localstack instances on a per-app basis, leading to duplication of code

* **What is the new behavior (if this is a feature change)?**
  Adding a basic Localstack commodity to the common dev-env for use by all applications that require a Localstack instance to test S3 connections against.

  Additionally, adding the new alias `localstack` to run `aws` commands directly against the container without having to bashin to the container and run them from there (similar to how the `manage` command allows you to skip that step). Note: I have aliased this to `awslocal ${@:1}` in the container (this is the same as `aws` but negating the need for the `--endpoint-url` parameter - commands run in the same way, however).

* **Does this PR introduce a breaking change? If so, what actions will users need to take in order to regain compatibility?**
  No; new commodity only

### All submissions

* [x] Have you followed the guidelines in our [Contributing](CONTRIBUTING.md) document?
* [x] Is this PR targeting the `develop` branch, and the branch rebased with commits squashed if needed?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New feature and bug fix submissions

* [x] Has your submission been tested locally?
* [x] Has documentation such as the [README](/README.md) been updated if necessary?
* [x] Have you linted your new code (using rubocop and markdownlint), and are not introducing any new offenses?

### CODE REVIEW

File: scripts/provision_localstack.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  if new_containers.include?('localstack')
+    new_db_container = true
+    puts colorize_yellow('The Localstack container has been newly created - '\
+                         'provision status in .commodities will be ignored')
+  end
+
+  localstack_initialised = false
+
+  config['applications'].each do |appname, _appconfig|
+    # To help enforce the accuracy of the app's dependency file, only search for init sql
```
Comment (sichapman) [inline scripts/provision_localstack.rb:23]: sql?

File: scripts/add-aliases.sh [hunk 2]
```diff
@@ -88,6 +88,10 @@ function manage(){
     ex ${1} python3 manage.py ${@:2}
 }
 
+function localstack(){
```
Comment (sichapman) [inline scripts/add-aliases.sh:91]: Confirm - there are no command line utilities also called localstack that people may have that this alias may conflict with?

File: scripts/docker/localstack/docker-compose-fragment.3.7.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+version: '3.7'
```
Comment (sichapman) [inline scripts/docker/localstack/docker-compose-fragment.3.7.yml:1]: Could you also add a Compose Spec file (compose-fragment.yml) as support for that will be in next release and all commodities will need one

**Review by sichapman — CHANGES_REQUESTED** [review]

File: scripts/add-aliases.sh (hunk 2 above)
Comment (brooks630) [reply scripts/add-aliases.sh:91]: I had a look and couldn't see anything that should conflict with this. If it's running locally on the host machine it uses the `awslocal` alias.

**Pull request merged**